### DISCUSSION TITLE
Uploading Files, Speed up/Slow Down, Mute, solo and volume all missing from my version

### DISCUSSION THREAD

**Discussion by jamie2364** [CoderLine/alphaTab#1257] [Q&A] [answered]
Hey Daniel, first of all wow, what an incredible contribution you have made to the music development community. Thank you for your great work! :) If you're not told enough, you are really, really appreciated!

I've got alphatab working on my app but I'm not seeing the 'open file' feature where users can upload their own files, nor am I seeing the Playback Speed Settings, or Mute/Solo or volume on tracks. I installed this using your tutorial - should these features be included in there or am I missing something? I can't find anything in your docs about this. 

Thanks so much again and all the best with your future work! :)

**Comment by Danielku15** [comment, marked as answer]
The problem might be behind this statement: "I installed this using your tutorial". The tutorial, as the name says tries to teach developers how to use alphaTab to build their own software. alphaTab is an SDK (a library for developers), not (yet) a drop-in component shipping all functionalities in a customizable look&feel. It is up to you to build a component with the UI as you want to have it for your application. 

https://www.alphatab.net/docs/tutorial-web/introduction
![image](https://github.com/CoderLine/alphaTab/assets/674916/0726d142-b573-4945-96e7-4ad6007fda1a)

https://www.alphatab.net/docs/tutorial-web/conclusion
![image](https://github.com/CoderLine/alphaTab/assets/674916/1abdd7f1-a277-4700-9420-acd4d457f1fa)

**Comment by jamie2364** [comment]
Thank you so much for your response Daniel. I appreciate that and I understand what you mean about the library. However, I thought there might be some documentation on how those particular features were done on the homepage. Is there any insight on how they were done? I see some of them were included in the API documentation. Is that how it is acheived?

**Reply by Danielku15** [reply]
The idea is that you tackle it like this: 

1. You know you wanna add the speed up/slow down functionality. 
2. You build the UI component for it in the way you want it to look like. 
3. You go to the [reference docs](https://www.alphatab.net/docs/reference/api) and search for the related functionality. e.g. in this case `speed` is a good keyword and you will find the related function or property which controls it: https://www.alphatab.net/docs/reference/api/playbackspeed/
4. You call the related function/ set the related properties. 
5. You test the functionality. 

This is the principle that the tutorial tries to transport. 

What is indeed missing as of today, is an overall functional overview of alphaTab what is available on feature level. I haven't found/decided yet a format on how this could be documented best. Therefore the best overview is the API reference listing out all the operations. If you go through the available APIs it should tell you how to do things. 

The "how to use them" is up to you as developer / software engineer to decide and integrate.
